src/scripts/main.js

```diff
@@ -1,3 +1,391 @@
+inputName.setAttribute('name', 'name');
+
+inputName.setAttribute('type', 'text');
+inputName.required = true;
+
+labelName.appendChild(inputName);
+
+inputPosition.setAttribute('data-qa', 'position');
+inputPosition.setAttribute('name', 'position');
+
+inputPosition.required = true;
+inputPosition.setAttribute('type', 'text');
+labelPosition.appendChild(inputPosition);
+
+selectOffice.setAttribute('data-qa', 'office');
+selectOffice.setAttribute('name', 'office');
+
+const optionsTokyo = document.createElement('option');
+const optionsSingapore = document.createElement('option');
+const optionsLondon = document.createElement('option');
+const optionsNewYork = document.createElement('option');
+const optionsEdinburgh = document.createElement('option');
+const optionsSanFrancisco = document.createElement('option');
+
+optionsTokyo.textContent = 'Tokyo';
+optionsTokyo.setAttribute('value', 'Tokyo');
+optionsSingapore.textContent = 'Singapore';
+optionsSingapore.setAttribute('value', 'Singapore');
+
+optionsLondon.textContent = 'London';
+optionsLondon.setAttribute('value', 'London');
+
+optionsNewYork.textContent = 'New York';
+optionsNewYork.setAttribute('value', 'New York');
+
+optionsEdinburgh.textContent = 'Edinburgh';
+optionsEdinburgh.setAttribute('value', 'Edinburgh');
+
+optionsSanFrancisco.textContent = 'San Francisco';
+optionsSanFrancisco.setAttribute('value', 'San Francisco');
+
+selectOffice.append(
+  optionsTokyo,
+  optionsSingapore,
+  optionsLondon,
+  optionsNewYork,
+  optionsEdinburgh,
+  optionsSanFrancisco,
+);
+
+labelOffice.append(selectOffice);
+selectOffice.required = true;
+
+inputAge.setAttribute('data-qa', 'age');
+inputAge.setAttribute('name', 'age');
+
+inputAge.setAttribute('type', 'number');
+labelAge.appendChild(inputAge);
+inputAge.required = true;
+
+inputSalary.setAttribute('data-qa', 'salary');
+inputSalary.setAttribute('name', 'salary');
+inputSalary.setAttribute('type', 'number');
+inputSalary.required = true;
+labelSalary.appendChild(inputSalary);
+
+buttonSubmit.textContent = 'Save to table';
+buttonSubmit.setAttribute('type', 'submit');
+form.appendChild(labelName);
+form.appendChild(labelPosition);
+form.appendChild(labelOffice);
+form.appendChild(labelAge);
+form.appendChild(labelSalary);
+form.appendChild(buttonSubmit);
+form.setAttribute('type', 'submit');
+
+document.body.append(form);
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const newRows = document.createElement('tr');
+  const newName = document.createElement('td');
+  const newPosition = document.createElement('td');
+  const newOffice = document.createElement('td');
+  const newAge = document.createElement('td');
+  const newSalary = document.createElement('td');
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.setAttribute('class', 'notification');
+
+  div.style.position = 'absolute';
+
+  const valid = checkValidValueError();
```

Comment:
The function 'checkValidValueError' is used before its declaration. It's recommended to declare functions before they are used to improve code readability and prevent potential hoisting issues.